api/web3server_trace.go

```diff
@@ -0,0 +1,159 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/eth/tracers/logger"
+	"github.com/iotexproject/go-pkgs/hash"
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/action/protocol"
+	accountutil "github.com/iotexproject/iotex-core/action/protocol/account/util"
+	apitypes "github.com/iotexproject/iotex-core/api/types"
+	"github.com/iotexproject/iotex-core/blockchain/genesis"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/tidwall/gjson"
+)
+
+// fromLoggerStructLogs converts logger.StructLog to apitypes.StructLog
+func fromLoggerStructLogs(logs []logger.StructLog) []apitypes.StructLog {
+	ret := make([]apitypes.StructLog, len(logs))
+	for index, log := range logs {
+		ret[index] = apitypes.StructLog{
+			Pc:            log.Pc,
+			Op:            log.Op,
+			Gas:           math.HexOrDecimal64(log.Gas),
+			GasCost:       math.HexOrDecimal64(log.GasCost),
+			Memory:        log.Memory,
+			MemorySize:    log.MemorySize,
+			Stack:         log.Stack,
+			ReturnData:    log.ReturnData,
+			Storage:       log.Storage,
+			Depth:         log.Depth,
+			RefundCounter: log.RefundCounter,
+			OpName:        log.OpName(),
+			ErrorString:   log.ErrorString(),
+		}
+	}
+	return ret
+}
+
+func (svr *web3Handler) traceCall(ctx context.Context, in *gjson.Result) (interface{}, error) {
```

Comment:
keep all funcs of one class under the same file, or a new subclass could be created for debugging.

Reply:
done